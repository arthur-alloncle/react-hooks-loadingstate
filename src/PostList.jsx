import React from 'react';
import useApi from './common/hooks/useApi.jsx';
import { apiDataStates } from './common/constants.js';

const PostList = ({ title }) => {
  const { state, error, data, reload } = useApi(
    'https://api.nico.dev/2PACX-1vSNjZItcRIaqBeN8xIBQNjphBUqgBEOo149_bUjFMLOGsByT0LXqaBF3C-zN44ThrDeEdB5Q_bJsW5B/?row=true&table=532894058&reverse=true'
  );
  return (
    <div>
      <h1>{title}</h1>
      {state === apiDataStates.LOADING && (
        <div>
          <p>loading..</p>
        </div>
      )}
      {state === apiDataStates.ERROR && (
        <div>
          <p>
            <b>ERROR:</b>
            <br />
            {error || 'General error'}
          </p>
        </div>
      )}
      {state === apiDataStates.SUCCESS && (
        <ul>
          {data.map((item) => (
            <li key={item.link}>
              <b>{item.title}</b>
              <br />
              <a href={item.link} target="_blank">
                {item.link}
              </a>
            </li>
          ))}
        </ul>
      )}
      <button onClick={() => reload()}>reload</button>
    </div>
  );
};

export default PostList;
