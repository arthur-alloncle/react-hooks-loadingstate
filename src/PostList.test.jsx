import React from 'react';
import PostList from './PostList';
import renderer from 'react-test-renderer';

describe('PostList Snapshots', () => {
  it('loading renders correctly', () => {
    const tree = renderer.create(<PostList title="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('success renders correctly', () => {
    // need to find a way to mock posts
    const tree = renderer.create(<PostList title="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
