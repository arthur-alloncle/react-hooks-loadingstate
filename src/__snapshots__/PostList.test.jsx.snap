// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList Snapshots error renders correctly 1`] = `
<div>
  <h1>
    Test
  </h1>
  <div>
    <p>
      <b>
        ERROR:
      </b>
      <br />
      General Error
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    reload
  </button>
</div>
`;

exports[`PostList Snapshots loading renders correctly 1`] = `
<div>
  <h1>
    Test
  </h1>
  <div>
    <p>
      loading..
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    reload
  </button>
</div>
`;

exports[`PostList Snapshots success renders correctly 1`] = `
<div>
  <h1>
    Test
  </h1>
  <ul>
    <li>
      <b>
        Mocked Post 1
      </b>
      <br />
      <a
        href="https://css-tricks.com/creating-scheduled-push-notifications/"
        target="_blank"
      >
        https://css-tricks.com/creating-scheduled-push-notifications/
      </a>
    </li>
    <li>
      <b>
        Mocked Post 2
      </b>
      <br />
      <a
        href="https://www.cyon.ch/blog/Progressive-Web-Apps-PWA"
        target="_blank"
      >
        https://www.cyon.ch/blog/Progressive-Web-Apps-PWA
      </a>
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    reload
  </button>
</div>
`;
